src/components/ScrobbleItem.tsx

```diff
@@ -166,8 +167,12 @@ export default function ScrobbleItem({
 
   if (scrobble.timestamp) {
     if (!isToday(new Date(scrobble.timestamp))) {
-      timestampFormat = `${settings?.use12Hours ? 'M/d' : 'd/MM'} `;
+      timestampFormat = settings?.use12Hours ? 'M/d' : 'd/MM';
     }
+    if (getYear(new Date(scrobble.timestamp)) < getYear(new Date())) {
+      timestampFormat += '/yyyy';
+    }
+    timestampFormat += ' ';
     timestampFormat += settings?.use12Hours ? 'hh:mm a' : 'HH:mm';
     theTimestamp = format(new Date(scrobble.timestamp), timestampFormat);
```

Comment:
would probably be good to set the `new Date` part to its own variable to avoid the duplication
```suggestion
    scrobbleDate = new Date(scrobble.timestamp);
    if (!isToday(scrobbleDate)) {
      timestampFormat = settings?.use12Hours ? 'M/d' : 'd/MM';
    }
    if (getYear(scrobbleDate) < getYear(new Date())) {
      timestampFormat += '/yyyy';
    }
    timestampFormat += ' ';
    timestampFormat += settings?.use12Hours ? 'hh:mm a' : 'HH:mm';
    theTimestamp = format(scrobbleDate, timestampFormat);
```

Reply:
This is a good idea! 👍🏼